llvm/utils/TableGen/Common/CodeGenSchedule.cpp

```diff
@@ -1190,6 +1195,14 @@ void CodeGenSchedModels::collectProcItins() {
         LLVM_DEBUG(dbgs() << ProcModel.ItinsDef->getName()
                           << " missing class for itinerary "
                           << ItinDef->getName() << '\n');
+        if (EmitUnusedItinerary) {
+          IdxVec Writes, Reads;
+          unsigned SCIdx =
+              addSchedClass(ItinDef, Writes, Reads, /*ProcIndices*/ {0});
+          NumInstrSchedClasses++;
+          ProcModel.ItinDefList.resize(NumInstrSchedClasses);
+          ProcModel.ItinDefList[SCIdx] = ItinData;
+        }
```

Comment:
How about we instead extend `CodeGenSchedModels::collectSchedClasses` so it also reads `ItineraryRegPairs` and not just `Itinerary`? I think we can then have properly defined `Writes, Reads` variables as well if we call `addSchedClass` there. Essentially, we would call `addSchedClass` for `Inst->TheDef->getValueAsDef("Itinerary")` and then for every itinerary in `ItineraryRegPairs` as well.

Reply:
This does sounds like a better idea, 
I believe this way we can make `ItineraryRegPairs` (may be rename this to `ItineraryRegClassPairs`) part of standard llvm `Instruction` class.

Reply:
Have updated the CodeGenSchedule.cpp as suggested.